engine/dlib/src/dlib/dstrings.cpp

```diff
@@ -25,19 +25,15 @@
 
 int dmSnPrintf(char *buffer, size_t count, const char *format, ...)
 {
-    // MS-compliance
+    // mimics ms pre-ucrt vsnprintf_s behavior
     if (buffer == 0x0 || count == 0 || format == 0x0)
         return -1;
     va_list argp;
     va_start(argp, format);
-#if defined(_WIN32)
-    int result = _vsnprintf_s(buffer, count, _TRUNCATE, format, argp);
-#else
     int result = vsnprintf(buffer, count, format, argp);
-#endif
     va_end(argp);
-    // MS-compliance
-    if (count == 0 || (count > 0 && result >= (int)count))
+    // mimics ms pre-ucrt vsnprintf_s behavior
+    if (result >= (int)count)
```

Comment:
This is a `DMSDK`function you are changing, which affects plenty of extensions our there.
Since you did the chagne for a reason, I expect to see the tests for this?

And, you didn't comment _why_ the `count == 0` isn't good anymore?